react/week3/todolist/src/components/App.js

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import Form from './Form';
+import TodoList from './TodoList';
+import uuid from 'uuid';
+
+class App extends React.Component {
+   constructor() {
+      super();
+      this.state = {
+         isLoading: true,
+         todos: [],
+         id: uuid(),
+         done: true,
+         error: null,
+      };
+      this.handleSubmit = this.handleSubmit.bind(this);
+      this.handleChangeTodo = this.handleChangeTodo.bind(this);
+      this.handleChangeDate = this.handleChangeDate.bind(this);
+
+       this.handleCheckbox = this.handleCheckbox.bind(this);
+       this.removeTodo = this.removeTodo.bind(this);
+       this.handleEdit = this.handleEdit.bind(this);
+   }
```

Comment:
If you write all of your methods as arrow functions, you don't need to bind anything. 
eg: 
```js
handleChangeTodo(event) {
      this.setState({ description: event.target.value });
   }
```

would simply become: 

```js
handleChangeTodo = (event) => {
      this.setState({ description: event.target.value });
   }
```